src/Network/EventHandler.php

```diff
@@ -104,17 +104,29 @@ public static function process(string $id, string $payload): Response {
 			$response = Response::fromMessage($result->getStruct(), $request->format);
 		} catch (Throwable $e) {
 			Buddy::debug("[$id] processing error: {$e->getMessage()} in file {$e->getFile()}:{$e->getLine()}");
-			/** @var string $originalError */
-			$originalError = match (true) {
-				isset($request) => $request->error,
-				default => ((array)json_decode($payload, true))['error'] ?? '',
-			};
+			if (isset($request)) {
+				/** @var string $originalError */
+				$originalError = $request->error;
+				/** @var array<mixed> $originalErrorBody */
+				$originalErrorBody = $request->errorBody;
+			} else {
+				/** @var array{message?:array{error_body?:array<mixed>},error?:string} $payloadInfo */
+				$payloadInfo = (array)json_decode($payload, true);
+				if (isset($payloadInfo['error'])) {
+					$originalError = $payloadInfo['error']['message'];
+					$originalErrorBody = $payloadInfo['error']['body'] ?? [];
+				} else {
+					$originalError = '';
+					$originalErrorBody = [];
+				}
```

Comment:
If I get it right with isset we can remove if/else here and keep simple like this:

```suggestion
			$originalError = $payloadInfo['error']['message'] ?? '';
			$originalErrorBody = $payloadInfo['error']['body'] ?? [];
```

Reply:
Yes, I agree